packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -491,21 +503,74 @@ To edit platform code in an IDE see https://flutter.dev/developing-packages/#edi
 void _printPluginUpdatePubspecMessage(String pluginPath, String platformsString) {
   globals.printStatus('''
 You need to update $pluginPath/pubspec.yaml to support $platformsString.
-
 ''', emphasis: true, color: TerminalColor.red);
 }
 
 void _printNoPluginMessage() {
     globals.printError('''
 You've created a plugin project that doesn't yet support any platforms.
-
 ''');
 }
 
-void _printAddPlatformMessage(String pluginPath) {
+void _printPluginAddPlatformMessage(String pluginPath) {
   globals.printStatus('''
 To add platforms, run `flutter create -t plugin --platforms <platforms> .` under $pluginPath.
 For more information, see https://flutter.dev/go/plugin-platforms.
 
 ''');
 }
+
+/// returns a list of platforms which user has requested a platform and if it's not enabled
+List<String> _getPlatformWarningList(List<String> requestedPlatforms) {
+  final List<String> _disabledPlatform = <String>[
+  if (requestedPlatforms.contains('android') && !featureFlags.isAndroidEnabled)
+    'android',
+  if (requestedPlatforms.contains('ios') && !featureFlags.isIOSEnabled)
+    'ios',
+  if (requestedPlatforms.contains('web') && !featureFlags.isWebEnabled)
+    'web',
+  if (requestedPlatforms.contains('macos') && !featureFlags.isMacOSEnabled)
+    'macos',
+  if (requestedPlatforms.contains('windows') && !featureFlags.isWindowsEnabled)
+    'windows',
+  if (requestedPlatforms.contains('linux') && !featureFlags.isLinuxEnabled)
+    'linux',
+  ];
+
+  return _disabledPlatform;
+}
+
+void _printWarningDisabledPlatform(List<String> platforms, String relativePath) {
+  final  List<String> _mobile = <String>[], _desktop = <String>[], _web = <String>[];
+  for (final String platform in platforms) {
+    if (platform == 'android' || platform == 'ios') {
+      _mobile.add(platform);
+    } else if (platform == 'web') {
+      _web.add(platform);
+    } else if (platform == 'macos' || platform == 'windows' || platform == 'linux') {
+      _desktop.add(platform);
+    }
+  }
+
+  if (_mobile.isNotEmpty) {
+    globals.printStatus('''
+The mobile ${_mobile.length > 1 ? 'platforms' : 'platform'}: ${_mobile.join(', ')} ${_mobile.length > 1 ? 'are' : 'is'} not enabled.
+Enable them and then
+
+  \$ cd $relativePath
+  \$ flutter create .
+''');
+  }
+  if (_desktop.isNotEmpty) {
+    globals.printStatus('''
+The desktop ${_desktop.length > 1 ? 'platforms' : 'platform'}: ${_desktop.join(', ')} ${_desktop.length > 1 ? 'are' : 'is'} not supported on your local environment.
+For more details, see: https://flutter.dev/desktop
```

Comment:
@csells could you double check the message here?
Basically in `flutter create`, if the user ask for desktop platforms but the desktop features are not enabled, we print this message to the user:
```
The desktop platform windows is not supported on your local environment.
For more details, see: https://flutter.dev/desktop
```

Below we have something similar for web.